bin/helpers/utils.js

```diff
@@ -1286,6 +1290,32 @@ exports.setConfig = (bsConfig, args) => {
   }
 }
 
+// set configs if enforce_settings is passed
+exports.setEnforceSettingsConfig = (bsConfig) => {
+  let config_args = (!this.isUndefined(bsConfig) && !this.isUndefined(bsConfig.run_settings) && !this.isUndefined(bsConfig.run_settings.config)) ? bsConfig.run_settings.config : undefined;
+  if ( this.isUndefined(config_args) || !config_args.includes("video") ) {
+    let video_args = (this.isUndefined(bsConfig.run_settings.video_config) || this.isUndefined(bsConfig.run_settings.video_config.video) || !bsConfig.run_settings.video_config.video ) ? 'video=false' : 'video=true' ;
+    video_args += (this.isUndefined(bsConfig.run_settings.video_config) || this.isUndefined(bsConfig.run_settings.video_config.videoUploadOnPasses) || !bsConfig.run_settings.video_config.videoUploadOnPasses ) ? ',videoUploadOnPasses=false' : ',videoUploadOnPasses=true';
+    config_args = this.isUndefined(config_args) ? video_args : config_args + ',' + video_args;
+    logger.debug(`Setting video_args for enforce_settings to ${video_args}`);
+  }
+  if ( (!this.isUndefined(bsConfig) && !this.isUndefined(bsConfig.run_settings) && !this.isUndefined(bsConfig.run_settings.baseUrl)) && (this.isUndefined(config_args) || !config_args.includes("baseUrl")) ) {
+    let base_url_args = 'baseUrl='+bsConfig.run_settings.baseUrl;
+    config_args = this.isUndefined(config_args) ? base_url_args : config_args + ',' + base_url_args;
+    logger.debug(`Setting base_url_args for enforce_settings to ${base_url_args}`);
+  }
+  // set specs in config of specpattern to override cypress config
+  if(!this.isUndefined(bsConfig) && !this.isUndefined(bsConfig.run_settings) && !this.isUndefined(bsConfig.run_settings.specs)) {
+    let spec_pattern_args = 'specPattern="'+bsConfig.run_settings.specs+'"';
+    if( bsConfig.run_settings.cypressTestSuiteType !== Constants.CYPRESS_V10_AND_ABOVE_TYPE) {
+      spec_pattern_args = 'testFiles="'+bsConfig.run_settings.specs+'"';
+    }
+    config_args = this.isUndefined(config_args) ? spec_pattern_args : config_args + ',' + spec_pattern_args;
+  }
+  if ( !this.isUndefined(config_args) ) bsConfig["run_settings"]["config"] = config_args;
```

Comment:
Probable can create a helper method `const isDefined = val => !this.isUndefined(val)` and use it everywhere